extensions/flyway/deployment/src/main/java/io/quarkus/flyway/deployment/FlywayProcessor.java

```diff
@@ -84,6 +88,22 @@ class FlywayProcessor {
 
     private static final Logger LOGGER = Logger.getLogger(FlywayProcessor.class);
 
+    @BuildStep
+    void preventScannerRegistration(BuildProducer<RemovedResourceBuildItem> producer) {
+        if (!QuarkusClassLoader.isApplicationClass("software.amazon.awssdk.services.s3.S3Client")) {
```

Comment:
Will `isApplicationClass` be true for a dependency? I would have thought you needed `isClassPresentAtRuntime` instead?